src/Service.Tests/CosmosTests/SchemaGeneratorFactoryTests.cs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Serialization;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Generator;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Azure.DataApiBuilder.Service.Tests.CosmosTests
+{
+    /// <summary>
+    /// Contains unit tests for SchemaGeneratorFactory related to Cosmos DB NoSQL.
+    /// </summary>
+    [TestClass, TestCategory(TestCategory.COSMOSDBNOSQL)]
+    public class SchemaGeneratorFactoryTests
+    {
+        /// <summary>
+        /// Tests the ExportGraphQLFromCosmosDB method to ensure it generates a GraphQL schema successfully from Cosmos DB.
+        /// </summary>
+        [TestMethod]
+        [DataRow("noop-database", "Container0", "mydb1", "container1", null, "Container0,Container1,Container2", DisplayName = "Test with global and entity level database and container names")]
+        [DataRow(null, "Container0", "mydb1", "container1", "Connection String and Database name must be provided in the config file", null, DisplayName = "Test with missing global database name")]
+        [DataRow("noop-database", null, "mydb1", "container1", "Unexpected Source format", null, DisplayName = "Test with missing global container name")]
```

Comment:
if global container is missing, wouldnt you rely on entity's container name to be sufficient? So, here it should be `container1`..

Reply:
added doc